### PR TITLE
Feature: Add MCP publish-document ability [ED-25127]

### PR DESCRIPTION
**Pull request by IshayMaya**
## Summary
- Adds Core MCP ability `elementor/publish-document` to transition any Elementor document (page, post, theme template, popup) from any status to `publish`.
- Uses `current_user_can( 'publish_post', $post_id )` for per-document permission checks and is idempotent when the document is already published.
- Registers the tool on the Elementor MCP server and adds PHPUnit coverage for validation, permissions, publish transition, idempotency, and failure handling.

## Test plan
- [ ] Run `tests/phpunit/elementor/modules/mcp/test-publish-document-ability.php` in the full PHPUnit environment
- [ ] Verify MCP server exposes `elementor/publish-document`
- [ ] Publish a draft page via MCP and confirm it appears on the front end
- [ ] Confirm calling publish on an already-published document returns success without error
- [ ] Confirm users without `publish_post` receive a 403

Made with [Cursor](https://cursor.com)

## 1. Problem & Context

Adds publish capability to Elementor's MCP (Model Context Protocol) abilities, enabling AI agents to transition documents from draft→publish after edits complete. Idempotent to handle already-published documents gracefully.

## 2. What Changed (Where)

- New `Publish_Document_Ability` class with validation, permission checks, and error handling
- Module registration in `mcp/module.php` with ability instantiation and server tools list
- Comprehensive test suite covering validation, authorization, state transitions, and edge cases

## 3. How It Works

Input validation (post_id required, reject revisions) → permission check (`publish_post` cap against main_id) → idempotency check (already published = no-op success) → `wp_update_post()` to publish → response with document metadata and helper links. Main_id resolution via `$document->get_main_id()` handles document remapping correctly.

## 4. Risks

None identified. Authorization uses `publish_post` cap (correct), checks against remapped main_id not input (guarded by test), and error handling is defensive. Idempotent design prevents duplicate publishes from causing issues.

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>

### CODE REVIEW

File: modules/mcp/abilities/publish-document-ability.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+		if ( ! $post_id ) {
+			return new \WP_Error(
+				'invalid_post_id',
+				__( 'A valid post_id is required.', 'elementor' ),
+				[ 'status' => \WP_Http::BAD_REQUEST ]
+			);
+		}
+
+		if ( wp_is_post_revision( $post_id ) ) {
```
Comment (IshayMaya) [inline modules/mcp/abilities/publish-document-ability.php:67]: @Omerisra6 WDYT? i didn't see we do this check usually but it came up after initially i just always took the main document id (see in line 85) and it looked like we can have a mismatch in the case of revisions

Comment (Omerisra6) [reply modules/mcp/abilities/publish-document-ability.php:67]: Lets keep it

**Review by Omerisra6 — DISMISSED** [review]

**Review by Omerisra6 — APPROVED** [review]
